src/Tracy/Bar.php

```diff
@@ -17,12 +17,36 @@
  */
 class Bar
 {
+	/** @var string */
+	public $directory;
+
 	/** @var string[] */
 	public $info = array();
 
 	/** @var IBarPanel[] */
 	private $panels = array();
 
+	/** @var resource */
+	private $handle;
+
+	/** @var string */
+	private $id;
+
+
+	public function register()
+	{
+		if (isset($_GET['_tracy_bar'])) {
+			$this->dispatch($_GET['_tracy_bar']);
+			exit;
+		}
+
+		$this->id = md5(uniqid(TRUE));
+		$this->handle = fopen(($this->directory ?: ini_get('upload_tmp_dir')) . '/bar.' . $this->id, 'w');
+		flock($this->handle, LOCK_EX);
```

Comment:
I already know! The other part (currently not published) of this PR, support for AJAX, requires locking, but here it is useless